### PR TITLE
Refactored register_preamble

### PR DESCRIPTION
**Pull request by anutosh491**
# Description

Addressed the [FIXME](https://github.com/compiler-research/xeus-cpp/blob/f071df023cb671df101b7176245acd8e97059806/src/xinterpreter.cpp#L394) comment in the code base. 

## Type of change

Please tick all options which are relevant.

- [ ] Bug fix
- [ ] New feature
- [ ] Added/removed dependencies
- [ ] Required documentation updates

### CODE REVIEW

**Comment by anutosh491** [comment]
@vgvassilev not sure if the above clang tidy suggestions should be applied. If we could use nodiscard but we can also do without it I suppose.

File: test/test_interpreter.cpp [hunk 1]
```diff
@@ -651,9 +650,9 @@ TEST_SUITE("xsystem_clone")
     {
         xcpp::xsystem system;
 
-        xcpp::xpreamble* clone = system.clone();
+        std::unique_ptr<xcpp::xpreamble> clone = system.clone();
 
-        REQUIRE(dynamic_cast<xcpp::xsystem*>(clone) != nullptr);
+        REQUIRE(dynamic_cast<xcpp::xsystem*>(clone.get()) != nullptr);
```
Comment (vgvassilev) [inline test/test_interpreter.cpp:655]: Does `xsystem` have multiple inheritance? If not we do not need `dynamic_cast`s here.

Comment (anutosh491) [reply test/test_interpreter.cpp]: Addressed.

**Review by vgvassilev — APPROVED** [review]
LGTM!

**Pull request merged**